shopinvader/models/res_partner.py

```diff
@@ -55,6 +55,37 @@ def _is_partner_duplicate_prevented(self):
         get_param = self.env["ir.config_parameter"].sudo().get_param
         return str2bool(get_param("shopinvader.no_partner_duplicate"))
 
+    @api.constrains("email", "shopinvader_bind_ids")
+    def _check_unique_email(self):
+        if not self._is_partner_duplicate_prevented():
+            return True
+        self.env["res.partner"].flush(["email", "shopinvader_bind_ids"])
+        self.env.cr.execute(
+            """
+            SELECT
+                email
+            FROM (
+                SELECT
+                    rp.id,
+                    rp.email,
+                    ROW_NUMBER() OVER (PARTITION BY rp.email) AS Row
+                FROM
+                    res_partner rp
+                WHERE rp.email is not null
+                    and rp.active = True
+                    and EXISTS (SELECT FROM shopinvader_partner sp WHERE sp.record_id = rp.id)
```

Comment:
what about `AND sp.active`?

Reply:
shopinvader_partner doesn't have the active field, we have is_shopinvader_active but it is made computable so cannot be used in sql